### PR TITLE
Use higher trajectory and rotate gripper for entering large object in bin

### LINKED ISSUE #1676: Big object get stuck during entering bin
- [x] approach from upper side
- [x] give some angle during entering bin

https://drive.google.com/file/d/0B5DV6gwLHtyJOVVtU2plb2t6STQ/view?usp=sharing

### PR DESCRIPTION
**Pull request by knorth55**
Closes #1676 

When robot grasp large object
- put the object in proper bin
- enter in bin with offset :z +50
- exit from bin with offset :z 0

When exiting from bin
- rotate gripper to 0

### CODE REVIEW

File: jsk_2016_01_baxter_apc/euslisp/main-stow.l [hunk 1]
```diff
@@ -118,7 +118,11 @@
               (setq hcv (send *ri* :get-val '_hard-coded-variables))
               (if (and object-length (> 0.20 object-length))
                 (setq entrance-x-list (list -200 -150 -100 -50 0 150))
-                (setq entrance-x-list (list -200 -150 -100 -50 0 150 200)))
+                (progn
+                  (setq entrance-x-list (list -200 -150 -100 -50 0 150 200))
+                  (if (find target-bin '(:a :b :c :j :k :l))
+                    (setq hcv 0)
```
Comment (pazeshun) [inline jsk_2016_01_baxter_apc/euslisp/main-stow.l]: hcv is hash table

File: jsk_2016_01_baxter_apc/euslisp/main-stow.l [hunk 2]
```diff
@@ -115,15 +115,19 @@
...
+                (progn
+                  (setq entrance-x-list (list -200 -150 -100 -50 0 150 200))
+                  (setq offset-avoid-bin-top 0)
+                  (send *ri* :angle-vector (send *baxter* :rotate-gripper arm 45 :relative nil))))
               (setq avs-picked->place-bin
                     (mapcar
                       #'(lambda (x)
                           (send *ri* :ik->bin-entrance arm target-bin
-                                :offset (float-vector x 0 (gethash :offset-avoid-bin-top hcv))))
+                                :offset (float-vector x 0 offset-avoid-bin-top)))
```
Comment (pazeshun) [inline jsk_2016_01_baxter_apc/euslisp/main-stow.l]: Use `gripper-angle` and `rotation-axis` key to reflect gripper angle

Comment (knorth55) [inline jsk_2016_01_baxter_apc/euslisp/main-stow.l]: you mean `:ik->bin-entrance`? 
you want to modify in `main.l`, too?

Comment (pazeshun) [inline jsk_2016_01_baxter_apc/euslisp/main-stow.l]: No. You change gripper angle at https://github.com/knorth55/jsk_apc/blob/21d34ab98d90d15453aca61027e0909479cad2c8/jsk_2016_01_baxter_apc/euslisp/main-stow.l#L125, but gripper angle becomes 90 degrees in `:ik->bin-entrance` without `gripper-angle` key

Comment (knorth55) [inline jsk_2016_01_baxter_apc/euslisp/main-stow.l]: i got what you mean.
`:rotation-axis :z` is good, no?

File: jsk_2016_01_baxter_apc/euslisp/main-stow.l [hunk 3]
```diff
@@ -115,23 +115,42 @@
...
-              (setq hcv (send *ri* :get-val '_hard-coded-variables))
-              (if (and object-length (> 0.20 object-length))
-                (setq entrance-x-list (list -200 -150 -100 -50 0 150))
-                (setq entrance-x-list (list -200 -150 -100 -50 0 150 200)))
+              (setq offset-avoid-bin-top
+                    (gethash :offset-avoid-bin-top (send *ri* :get-val '_hard-coded-variables)))
+              (if (and object-length (> object-length 0.2))
+                (progn
+                  (setq entrance-x-list (list -200 -150 -100 -50 0 150 200))
+                  (setq offset-avoid-bin-top 50)
```
Comment (pazeshun) [inline jsk_2016_01_baxter_apc/euslisp/main-stow.l]: Is this offset ok for Bin d, e, f, g, h, i?
If ok, don't worry about this comment.

**Comment by pazeshun** [comment]
If this was tested to all bins, +1.

**Comment by pazeshun** [comment]
It conflicts after #1702 was merged

File: jsk_2016_01_baxter_apc/euslisp/main-stow.l (hunk 3 above)
Comment (knorth55) [inline jsk_2016_01_baxter_apc/euslisp/main-stow.l]: i only tested for bin a b c j k l.
But in this PR, it only select these bins.

Comment (pazeshun) [inline jsk_2016_01_baxter_apc/euslisp/main-stow.l]: ok

**Comment by pazeshun** [comment]
+1

**Pull request merged**